### PR TITLE
feat: check that go module name is valid and not a keyword

### PR DESCRIPTION
**Pull request by michaelneale**
Fix for: https://github.com/TBD54566975/ftl/issues/1710

### CODE REVIEW

**Comment by michaelneale** [comment]
still WIP

**Review by alecthomas — APPROVED** [review]
Awesome, thanks Mic!

File: cmd/ftl/cmd_init.go [hunk 1]
```diff
@@ -36,13 +38,31 @@ type initGoCmd struct {
...
+	if !isValidGoPackageName(i.Name) {
+		return fmt.Errorf("module name %q is not a valid Go package name", i.Name)
+	}
+
 	if !schema.ValidateName(i.Name) {
 		return fmt.Errorf("module name %q is invalid", i.Name)
 	}
+
+	// Check if the module name is a Go keyword
+	if token.Lookup(i.Name).IsKeyword() {
```
Comment (alecthomas) [inline cmd/ftl/cmd_init.go:62]: Maybe fold this into the function above?

File: cmd/ftl/cmd_init.go [hunk 2]
```diff
@@ -36,13 +38,31 @@ type initGoCmd struct {
 	Name    string            `arg:"" help:"Name of the FTL module to create underneath the base directory."`
 }
 
+func isValidGoPackageName(name string) bool {
+	validNamePattern := `^[a-zA-Z][a-zA-Z0-9_]*$`
+	matched, _ := regexp.MatchString(validNamePattern, name)
```
Comment (alecthomas) [inline cmd/ftl/cmd_init.go:43]: Use `regexp.MustCompile()` instead of ignoring the error.

**Review by matt2e — APPROVED** [review]

**Pull request merged**